components/shared/process/channel.rs

```diff
@@ -0,0 +1,185 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! Enum wrappers to be able to select different channel implementations at runtime.
+
+use std::fmt;
+
+use ipc_channel::router::ROUTER;
+use serde::{Deserialize, Serialize};
+use servo_config::opts;
+
+#[derive(Deserialize, Serialize)]
+pub enum Sender<T: Serialize> {
+    Ipc(ipc::Sender<T>),
+    Mpmc(mpmc::Sender<T>),
+}
+
+impl<T> Clone for Sender<T>
+where
+    T: Serialize,
+{
+    fn clone(&self) -> Self {
+        match *self {
+            Sender::Ipc(ref chan) => Sender::Ipc(chan.clone()),
+            Sender::Mpmc(ref chan) => Sender::Mpmc(chan.clone()),
+        }
+    }
+}
+
+impl<T: Serialize> fmt::Debug for Sender<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Sender(..)")
+    }
+}
+
+impl<T: Serialize> Sender<T> {
+    #[inline]
+    pub fn send(&self, msg: T) -> SendResult {
+        match *self {
+            Sender::Ipc(ref sender) => sender.send(msg).map_err(|_| SendError),
+            Sender::Mpmc(ref sender) => sender.send(msg).map_err(|_| SendError),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SendError;
+pub type SendResult = Result<(), SendError>;
+
+#[derive(Debug)]
+pub struct ReceiveError;
+pub type ReceiveResult<T> = Result<T, ReceiveError>;
+
+pub enum Receiver<T>
+where
+    T: for<'de> Deserialize<'de> + Serialize,
+{
+    Ipc(ipc::Receiver<T>),
```

Comment:
Just use ipc_channel directly:

```suggestion
    Ipc(ipc_channel::ipc::IpcReceiver<T>),
```

Reply:
Fixed in [731b64f](https://github.com/servo/servo/pull/32339/commits/731b64fa2fc597d0080905cb1a9e562c571f648b)